miio/integrations/petfeeder/status.py

```diff
@@ -0,0 +1,85 @@
+from datetime import timedelta
+from enum import Enum
+from typing import Any, Dict
+
+from miio import DeviceStatus
+
+
+class FoodLevel(Enum):
+    Normal = 0
+    Low = 1
+    Empty = 2
+
+
+class FoodOutletStatus(Enum):
+    Ok = 0
+    Blocked = 1
+
+
+class FoodStorageLid(Enum):
+    Closed = 0
+    Open = 1
+
+
+class PetFeederStatus(DeviceStatus):
+    """Container for status reports from the Pet Feeder / Smart Pet Food Dispenser."""
+
+    def __init__(self, data: Dict[str, Any]) -> None:
+        """
+        Response from pet feeder (mmgg.feeder.petfeeder):
+
+        {
+            "food_level": 0,
+            "feed_plan": 1,
+            "food_outlet": 0,
+            "undeclared": 0,
+            "clean_days_left": 11,
+            "food_storage_lid": 0,
+            "desiccant_days_left": 21,
+            "weight_level": 0,
+            "wifi_led": 0,
+            "button_lock": 1,
+            "country_code": 255
+        }
+        """
+        self.data = data
+
+    @property
+    def food_level(self) -> FoodLevel:
+        """Current food level / status."""
+        return FoodLevel(self.data["food_level"])
+
+    @property
+    def feed_plan(self) -> bool:
+        """Automatic feeding status."""
+        return bool(self.data["feed_plan"])
+
+    @property
+    def food_outlet(self) -> FoodOutletStatus:
+        """Food outlet status."""
+        return FoodOutletStatus(self.data["food_outlet"])
+
+    @property
+    def food_storage_lid(self) -> FoodStorageLid:
```

Comment:
`is_storage_door_open`? I'm not sure what these all signify, but I'd recommend using bools with some more descriptive names here :-)

Reply:
```lid - noun
a removable or hinged cover for the top of a container.```

It's a lid, not door 😸 

Again, makes sense to a device owner.

Reply:
![Xiaomi-Xiaowan-Smart-Pet-Feeder](https://user-images.githubusercontent.com/1288525/146284779-9a2d962b-924f-4328-8007-74eaf28827ee.png)

Reply:
So I was thinking about this from the perspective of downstream users, let's say this were integrated in homeassistant, it would still be a boolean on/off (or open/closed, or something similar), so IMO it makes sense to handle it as such.

Reply:
Oh, I thought you were just picking on the function name 😆 
Sure, as we said before it would be a binaryswitch with door device_class of open/closed for hass.

Thought just comes back to the CLI again as I know there are quite a few people for one reason or another who do jobs via scripts which just call miiocli rather than writing actual code.